tests/console/krb5.pm

```diff
@@ -0,0 +1,125 @@
+
+    #Fix realm address on krb5.cnf file:
+    assert_script_run "sed -i 's/openqa.kerberus.org.kerberus.org/openqa.kerberus.org/g' /etc/krb5.conf";
+
+    #Test Case 1439087: ssh with MIT Kerberos 5:
+    record_info 'Testcase 1439087:';
+
+    #kadmind starts the Kerberos administration server systemd unit:
+    systemctl('restart kadmind', expect_true => 1);
+    systemctl('stop kadmind',    expect_true => 1);
+    systemctl('start kadmind',   expect_true => 1);
+    systemctl('status kadmind',  expect_true => 1);
+
+    #krb5kdc is the Kerberos v.5 Auth. service and key distrib. center:
+    systemctl('restart krb5kdc', expect_true => 1);
+    systemctl('stop krb5kdc',    expect_true => 1);
+    systemctl('start krb5kdc',   expect_true => 1);
+    systemctl('status krb5kdc',  expect_true => 1);
+
+    script_run 'mkdir -p /run/user/1001/krb5cc';
+    script_run 'chown tester:users /run/user/1001/krb5cc';
+
+    #confirm we have no existing kinit tickets cache:
+    script_run 'su - tester', 0;
+    type_string "klist 2> /tmp/krb5\n";
+    script_run 'logout', 0;
+    validate_script_output "grep -iEc \"credentials|not|found|no\" /tmp/krb5", sub { /1/ };
+
+    #create kinit tickets cache for the user:
+    script_run 'su - tester', 0;
+    type_string "kinit\n";
+    #just a fast paced and harmless pwd:
+    type_string "1234wert\n";
+
+    #confirm the users tickets cache is created, also confirm its on the FQDN:
+    type_string "klist > /tmp/krb5-klist-cache 2> /dev/null\n";
+    script_run 'logout', 0;
+    validate_script_output "grep -c \"krbtgt\/openqa.kerberus.org\@openqa.kerberus.org\" /tmp/krb5-klist-cache", sub { /1/ };
+
+    #validate tickets cache output integrity:
+    validate_script_output 'wc -l /tmp/krb5-klist-cache', sub { /6/ };
+
+    #test the destroy kinit cache command:
+    script_run 'su - tester', 0;
+    type_string "kdestroy\n";
+    type_string "klist 2> /tmp/krb5\n";
+    script_run 'logout', 0;
+    validate_script_output "grep -iEc \"credentials|not|found|no\" /tmp/krb5", sub { /1/ };
+
+    #test the service management script:
+    assert_script_run 'rckadmind start';
+    assert_script_run 'rckadmind stop';
+    assert_script_run 'rckadmind start';
+    assert_script_run 'rckadmind restart';
+    assert_script_run 'rckadmind reload';
+    assert_script_run 'rckadmind force-reload';
+    validate_script_output "rckadmind status |grep -ic active", sub { /1/ };
+
+    #cleanup/restore VM original state:
+    script_run 'rm /tmp/krb5-klist-cache';
+    script_run 'rm -f /var/lib/kerberos/krb5kdc/principal*';
+    script_run 'mv /etc/hosts-bkp /etc/hosts';
+    script_run 'mv /etc/hostname-bkp /etc/hostname';
+    script_run 'rm /tmp/krb5';
+    script_run 'hostname `cat /tmp/hostname`';
+    script_run 'rm /tmp/hostname';
+
+    #important log that we have indeed returned former hostname:
+    record_info 'Restored Hostname';
+    assert_script_run 'hostname';
```

Comment:
Maybe use validate_script_output for the expected hostname ?

Reply:
Or store it at the beginning to variable and compare it here ?

Reply:
Sounds a good move!